test/functional/viml/completion_spec.lua

```diff
@@ -959,4 +959,34 @@ describe('completion', function()
       set complete&vim completeopt&vim
     ]])
   end)
+
+  it('CompleteChanged autocommand', function()
+    curbufmeths.set_lines(0, 1, false, { 'foo', 'bar', 'foobar', ''})
+    source([[
+      set complete=. completeopt=noinsert,noselect,menuone
+      autocmd! CompleteChanged * :let g:item = get(v:event, 'completeitem', v:null)
+      call cursor(4, 1)
+    ]])
+
+    feed('S')
+    wait()
+    feed('f')
+    wait()
+    feed('<C-N>')
+    wait()
+    eq({}, eval('g:item'))
+    feed('<C-N>')
+    wait()
+    eq('foo', eval('g:item["word"]'))
+    feed('<C-N>')
```

Comment:
@chemzqm  when press `<Up>` the complete item is changed, but do not insert completed word, then what is the g:item

Reply:
The current selected complete item, as you would expected.